### DISCUSSION TITLE
Trying to collect all the necessary data to a checkout page and display it properly

### DISCUSSION THREAD

**Discussion by henkegithub** [neuefische/questions#669] [Web]
[Here is the PR](https://github.com/henkegithub/instroke-app/pull/16)
I created two arrays of list items(products) and placed them on the order menu and now I am stuck on how to get the data from the Order-List to be displayed and updated correctly on the Overview Page.

**/Product/index.js:**
```
import { useState } from "react";

export function Product({ name, price }) {
  const [counter, setCounter] = useState(0);

  //increase counter
  const increase = () => {
    setCounter((count) => count + 1);
  };

  //decrease counter
  const decrease = () => {
    if (counter > 0) {
      setCounter((count) => count - 1);
    }
  };

  //reset counter
  const reset = () => {
    setCounter(0);
  };
  return (
    <>
      <li>{name}</li>
      <li>{price} €</li>
      <button className="control__btn" onClick={increase}>
        +
      </button>
      <p>{counter}</p>
      <button className="control__btn" onClick={decrease}>
        -
      </button>
      <button className="reset" onClick={reset}>
        Reset
      </button>
    </>
  );
}
```
**Order Menu/Drinks:**
```
import { Product } from "@/components/Product";
export default function Drinks() {
  const products = [
    {
      name: "Apfelschorle",
      price: 2.5,
    },
    {
      name: "Johannisbeersaft",
      price: 2.5,
    },
    {
      name: "Zitroneneistee",
      price: 3.5,
    },
    {
      name: "Pfirsicheistee",
      price: 3.5,
    },
  ];

  return (
    <ul>
      {products.map((product) => (
        <Product key={product} name={product.name} price={product.price} />
      ))}
    </ul>
  );
}
```

**Order Menu/Maindishes**
```
import { Product } from "@/components/Product";

export default function Maindishes() {
  const products = [
    {
      name: "Schnitzel",
      price: 10.5,
    },
    {
      name: "Spätzle",
      price: 9.5,
    },
    {
      name: "Rindergulasch",
      price: 18.5,
    },
    {
      name: "Sauerbraten",
      price: 14.5,
    },
  ];

  return (
    <ul>
      {products.map((product) => (
        <Product key={product} name={product.name} price={product.price} />
      ))}
    </ul>
  );
}
```

**OrderOverview**
```
export default function OrderOverview() {
  return (
    <>
      <ul>
        <li>Item1</li>
        <li>10.50€</li>
        <li>2</li>
        <li>Item2</li>
        <li>9.50€</li>
        <li>1</li>
      </ul>
      <h1>Total:</h1>
      <p>30.50€</p>
    </>
  );
}
```

**Comment by ThomasF85** [comment]
Hello,

you pasted your code as plain text into this questions thread. It's impossible to see all syntax errors since there is things (especially in the jsx) missing. If you post code snippets post them as code (using the <> icon), not as plain text.

Also please provide the following:
- A link to a PR or a single commit with the code if you have one
- A description of what is not working, what the errors are
- Error messages from the browser or your vs code terminal

**Reply by henkegithub** [reply]
@ThomasF85 I tried my best to describe the problem. 
So when I click "+" or "-" I need to run a function which updates the recent state with the new amount selected and displays it in the cart as a list like products.name products.price and a total amount of all items selected.

But I don't know how to make this work

**Comment by ThomasF85** [comment]
You have to lift up your state.
You want to manipulate state (the amount of ordered items) on one page and then show these changes in state on a different page (the overview page).
For this to work with local state, you need to lift up the state containing all information about the amount of all items ordered.
Your state should be in the `_app.js` and be passed down to the pages and components via props.

**Reply by henkegithub** [reply]
The count state or what kind of state are we talking about? 

how can I create the amount of ordered items with a stateHook on one page? 

and how can I show the changes on a different page?

did I do it correct by putting the arrays in the app.js file?

[PR update](https://github.com/henkegithub/instroke-app/pull/16/commits/b75d09e33db120d6e731d94daa06b5f745457f68)

**Reply by ThomasF85** [reply]
Putting the arrays into the _app.js alone does not help. You have to create a state in your _app.js component.

You have to create one or multiple states that hold all necessary information to display the desired information. This state has to be created in the root (_app.js). This state has to incorporate all of your dishes and drinks and the amount of them.
For example:

```
const [items, setItems] = useState([
   {id: "1", name: "dish 1", price: 7, amount: 0}, {id: "2", name: "dish 2", price: 9, amount: 0}, ...
])
```

you can hand this state down to your overview page, that can then display information of all items and their amounts.
You can create functions that manipulate this state (increase / decrease the amount of an item) in the _app.js.
You can hand these functions over to your pages and your product component via props. Then your product component can use these function to increase / decrease the amount of an item.

**Reply by henkegithub** [reply]
latest update of the pr: https://github.com/henkegithub/instroke-app/pull/16/commits/7700637e825026daae6a056059b5709995fa2713

I can add items now but in the order list it says for example 3 items and on the order overview just displays 2 items and when I go back to the order list it shows 2 items and also when I refresh the order overview page I get an error with .reduce
